pkg/runner/step_action_remote.go

```diff
@@ -50,6 +50,14 @@ func (sar *stepActionRemote) pre() common.Executor {
 				return nil
 			}
 
+			sar.remoteAction.URL = sar.RunContext.Config.GitHubInstance
+			for _, throughAction := range sar.RunContext.Config.ThroughAction {
+				if (sar.remoteAction.Org + "/" + sar.remoteAction.Repo) == throughAction {
+					sar.remoteAction.URL = "github.com"
+					github.Token = sar.RunContext.Config.ThroughActionToken
+				}
+			}
+
```

Comment:
first of all I want to say I have the exact same problem, I just started using act and I found this limitation in 5 mins using it :D so this is super useful. Even though I had never seen the code before, I would have approached the problem slightly different, I would have created a fallback mechanism, if you have the github-instance specified, then if the clone fails, then you can try as a backup github.com and if that fails as well then you fail the entire thing. Because assuming you have tens of actions in your workflows then you would end up having to specify a massive list, which doesn't seem super clean. Also this would open the possibility to have multiple "providers" essentially even though I don't see the use case for that...yet. Just my 2 cents

Reply:
essentially here https://github.com/nektos/act/blob/d4272bd9fe13566ab76e5ad091aedbded74ff4e2/pkg/common/git.go#L190

Reply:
Mmm... wouldn't checking for a 404 be enough?

Reply:
> I would have created a fallback mechanism

I created something like that in my action utility, well it is a pain to mirror all github actions to a fake GHES server.
- call github api `/repos/<owner>/<repo>/commits?sha=refs%2fheads%2fmaster&page=1&limit=1&per_page=1`, `refs%2fheads%2fmaster` is a url data encoded query component can be sha, ref, tag or full reference name
- the previous api call will respond with http 200 and the current sha to clone, seem to mirror the sha outputs seen on github
- other http status ask the api from github.com if `--github-connect` flag is specified
We can now log the exact sha which was used by an action, like actions/runner.
- If no github api responds with 200, just try to clone from the specified github-instance.

_Using act is not the most secure way of running actions_
I prefer act would download the archive via `<github-or-ghes-url>/<owner>/<repo>/archive/refs/heads/main.zip` / `/<owner>/<repo>/archive/refs/heads/main.tar.gz`, because a initial git clone may take a while and act stalls.

A `--though-action` flag seem to be harder to use in my point of view.

Reply:
right, I meant regarding the strategy I was proposing

Reply:
> Is that not what this PR does?

right, I meant regarding the strategy I was proposing